### PR TITLE
Faq

### PR DESCRIPTION
**Pull request by julien-truffaut**

### CODE REVIEW

File: docs/src/main/tut/faq.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+Since `index` is weaker than `at`, we can implement an instance of `Index` on more data structure than `At`. 
+For instance, `List` or `Vector` only have an instance of `Index` because there is no way to insert an element at an 
+arbitrary index of a sequence.
+
+Note: `root` is a trick to help type inference. Without it, we would get the following error
+
+```tut:fail
+index("two").set(0)(m) 
+```
```
Comment (NightRa) [inline docs/src/main/tut/faq.md]: Will `tut` give the error here?

**Comment by NightRa** [comment]
Looks great for a start on the FAQ! :+1:

File: docs/src/main/tut/faq.md (hunk 1 above)
Comment (julien-truffaut) [inline docs/src/main/tut/faq.md]: yes it does. There is a limitation on the `fail` modifier where you need to use only failing code that's why I had to split examples into two `tut`, one for import and one for failing code

**Comment by julien-truffaut** [comment]
Thanks, I will merge and update the website. You will see the result

**Pull request merged**